lambdas/functions/main.py

```diff
@@ -2,123 +2,155 @@
-    allowed_origin_list = [ 'https://sscheck.dev.umccr.org', 'https://sscheck.prod.umccr.org', 'https://sscheck.umccr.org' ]
-    
-    if origin in allowed_origin_list:
-        return_origin = origin
-    else:
-        return_origin = allowed_origin_list[0]
-        
-    response =  {
-        'statusCode': status_code,
-        'headers': {
-            'Access-Control-Allow-Headers': 'Content-Type',
-            'Access-Control-Allow-Origin': return_origin,
-            'Access-Control-Allow-Methods': 'OPTIONS,POST,GET',
-        },
-        'body': body
-    }
-
-    return response
-
+# Logging
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
 
 def lambda_handler(event, context):
+    """
+    Parameters
+    ----------
+    event : Object
+        An object of payload pass through the lambda
+    context : Object
+        [NOT_USED] a aws resource information
+
+    Return
+    ----------
+    error_message : str
+        any error message that stops the check
+
+    """
+    logger.info('Processing samplesheet check based on the following event.')
+    logger.info(json.dumps(event))
 
     try:
         origin = event["headers"]["origin"]
     except KeyError:
         origin = ""
-    
+    logger.info(f'Parsing origin header content: {origin}')
+
     auth_header = event["headers"]["Authorization"]
+    logger.info(f'Parsing auth_header header content: {auth_header}')
+
 
     body = event["body"].encode()
 
     # Parse contentType
     try:
         content_type = event["headers"]['Content-Type']
-    except:
+    except KeyError:
         content_type = event["headers"]['content-type']
     ct = "Content-Type: "+content_type+"\n"
 
-    msg = email.message_from_bytes(ct.encode()+body)    
+    msg = email.message_from_bytes(ct.encode()+body)
 
     multipart_content = {}
     # retrieving form-data
     for part in msg.get_payload():
         # checking if filename exist as a part of content-disposition header
-        if part.get_filename():
-            # fetching the filename
-            file_name = part.get_filename()
         multipart_content[part.get_param('name', header='content-disposition')] = part.get_payload(decode=True)
 
     file_data = multipart_content["file"]
     log_level = multipart_content["logLevel"].decode("utf-8")
 
+    # Log Path
+    log_path = LOG_DIRECTORY["samplesheet_check"]
+
     # Check if data input is correct
     if log_level in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"] == False:
-        body = construct_body(check_status="FAIL", error_message="IncorrectLogLevel")
-        response = construct_response(status_code=400, body=body)
+
+        # Logging
+        logger.info(f"Log Level selected is not recognized. '{log_level}' is not an option.")
+
+        # Return response
+        body = construct_body(check_status="FAIL", error_message="IncorrectLogLevel", log_path=log_path)
+        response = construct_response(status_code=200, body=body, origin=origin)
```

Comment:
Hmm... I think it's still 200 since it's a valid payload just that the content is incorrect? Initially, I put it as 40X but then will it be the same thing if the file we are checking had invalid content but the payload is correct? That happened the previous version where invalid sample sheet data had incorrect data (invalid commas) and return 40X status code, and the front end interpret it as a network error which shows the pop-up dialog saying 'an error has occurred'

But, Im happy to change it to a 40X status code if it makes more sense tho

Reply:
That's fine. It's what I meant with the syntax being correct.